### ISSUE TITLE
Inference on Custom Dataset

### ISSUE THREAD

**Issue by Sagnnik** [yuli0103/CGB-DM#8]
I wanted to do inference on my custom dataset. For that I did convert the dataset into the .csv format and wanted to look into the Dataset Creation.
I am looking at the `__getitem__` function in the `test_cond_dataset`
```python
def __getitem__(self, idx):
        img_inp = Image.open(self.inp[idx]).convert("RGB")
        width, height = img_inp.size
        img_sal = Image.open(self.sal[idx]).convert("L")
        img_sal_sub = Image.open(self.sal_sub[idx]).convert("L")
        img_sal_map = Image.fromarray(np.maximum(np.array(img_sal), np.array(img_sal_sub)))
        img_inp = self.transform_rgb(img_inp)
        img_sal_map = self.transform_l(img_sal_map)
        imgs = torch.concat([img_inp, img_sal_map])

        sliced_df_annotated = self.groups_annotated.get_group(self.poster_name[idx])
        sliced_df_salbox = self.groups_salbox.get_group(self.poster_name[idx])

        gt_box = torch.tensor(list(map(eval, sliced_df_annotated["box_elem"])))
        gt_cls = list(sliced_df_annotated["cls_elem"])

        sal_box = torch.tensor(list(map(eval, sliced_df_salbox["box_elem"])))
        sal_box = box_xyxy_to_cxcywh(sal_box)
        sal_box[::2] /= width
        sal_box[1::2] /= height

        label_cls = np.zeros((self.max_elem, self.num_class))
        label_box = np.zeros((self.max_elem, 4))
        # label_cls_gt = np.zeros((self.max_elem, 1))
        # label_box_gt = np.zeros((self.max_elem, 4))

        for i in range(len(gt_cls)):
            if i >= self.max_elem:
                break
            label_cls[i][int(gt_cls[i])] = 1
            label_box[i] = gt_box[i]
            if label_box[i][0] > label_box[i][2] or label_box[i][1] > label_box[i][3]:
                label_box[i][:2], label_box[i][2:] = label_box[i][2:], label_box[i][:2]
            label_box[i] = box_xyxy_to_cxcywh(torch.tensor(label_box[i]))
            label_box[i][::2] /= width
            label_box[i][1::2] /= height

        for i in range(len(gt_cls), self.max_elem):
            label_cls[i][0] = 1
            # label_cls_gt[i][0] = 0

        label = np.concatenate((label_cls, label_box), axis=1)
        # label_gt = np.concatenate((label_cls_gt, label_box_gt), axis=1)

        label[:, self.num_class:] = 2 * (label[:, self.num_class:] - 0.5)
        sal_box = 2 * (sal_box - 0.5)
        return imgs, torch.tensor(label).float(), sal_box.float()
```
It is basically creating a padded classes and bounding box tensors made from the ground truth and there is no masking involved (depending on the task). Are you masking the input in the diffusion model itself? If that's the case then can the masking be done beforehand say in the csv dataset itself?

**Comment by yuli0103** [comment]
Thank you for your interest. I have implemented masking operations in the diffusion model part. The masking can be implemented first in the CSV file, which should only require minor code modifications.
